atom-feed/src/main/java/org/openmrs/module/bedmanagement/atomfeed/BedManagementAdviceActivatorComponent.java

```diff
@@ -0,0 +1,55 @@
+package org.openmrs.module.bedmanagement.atomfeed;
+
+import org.aopalliance.aop.Advice;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.bedmanagement.BedManagementActivatorComponent;
+import org.openmrs.module.bedmanagement.service.BedManagementService;
+import org.openmrs.module.bedmanagement.service.BedTagMapService;
+
+/**
+ * Registers AOP advice when the context starts up. This is done instead of declaring advice in
+ * config.xml in order to allow it to be conditionally loaded
+ */
+@OpenmrsProfile(modules = { "openmrs-atomfeed:*" })
+public class BedManagementAdviceActivatorComponent implements BedManagementActivatorComponent {
```

Comment:
This is the main thing being introduced - a conditionally loaded component that adds/removes advice during the module activation lifecycle.  The declarations that were previously hard-coded into config.xml are move here to execute conditionally.

Reply:
It would be good to adopt this technique for the appointments module too. Right now, we're generating warnings and ignoring them. This is much cleaner.

Reply:
@ibacher see my comment on the ticket https://openmrs.atlassian.net/browse/BED-5 - this is already in place in the appointments module.  I fixed it for this reason back in March.  Maybe we need to update the version in the refapp?